develop/tutorials/articles/340-data-upgrades-and-verifiers/creating-an-upgrade-process-for-your-application.markdown

```diff
@@ -1,78 +1,79 @@
+![Figure 1: In a registrator class, the developer specifies a registration for each schema version upgrade. The upgrade steps handle the database updates.](../../images/data-upgrade-module-upgrade-architecture.png)
 
 Now that you've learned each upgrade component's purpose, you're ready to create
-your module's upgrade process. 
+an upgrade process for your module. 
 
 Here's what's involved:
 
 - **Specifying the schema version**
 
+- **Specifying to wait for upgrade completion**
+
 - **Declaring dependencies**
 
 - **Writing upgrade steps**
 
 - **Writing the registrator**
 
-Let's start creating the upgrade process.
-
-## Creating an Upgrade Process [](id=creating-an-upgrade-process)
-
-Setting a data schema's version is easy, so you might as well do it first. Then
-you'll need to specify a dependency on the portal-upgrade module, and any other
-modules you plan to use in your schema upgrade. The remainder of the tutorial
-explains how to create the module's upgrade components: upgrade steps and the
-registrator.
-
-First, let's specify the schema version.
+It's time to get started. 
 
-### Specifying the Schema Version [](id=specifying-the-schema-version)
+## Specifying the Schema Version [](id=specifying-the-schema-version)
 
-In your module's `bnd.bnd` file and use Service Builder, specify a 
-`Liferay-Require-SchemaVersion` header with the new schema version value. 
-Here's an example schema version header: 
+In your module's `bnd.bnd` file, specify a `Liferay-Require-SchemaVersion`
+header with the new schema version value. Here's an example schema version
+header for a module whose new schema is version `1.1.0`: 
 
         Liferay-Require-SchemaVersion: 1.1.0
 
 **Important**: If no `Liferay-Require-SchemaVersion` header is specified,
 @product@ considers the `Bundle-Version` header value to be the database schema
 version.
 
-Modules that don't use  Service Builder should wait for the upgrade steps to be executed, 
-so a @Reference should be created with the target:
-    * release.bundle.symbolic.name: bundle symbolic name of the module
-    * release.schema.version: current schema version of the module
+Next, you must make sure the module's upgrade is executed before it's used. 
 
-A example can be found in [`PageCommentsPortlet` class's](https://docs.liferay.com/portal/7.0/javadocs/modules/apps/collaboration/comment/com.liferay.comment.page.comments.web/com/liferay/comment/page/comments/web/internal/portlet/PageCommentsPortlet.html)
+## Specifying to Wait for Upgrade Completion [](id=specifying-to-wait-for-upgrade-completion)
 
-public class PageCommentsPortlet extends MVCPortlet {
+Modules that don't use [Service Builder](https://dev.liferay.com/develop/tutorials/-/knowledge_base/7-0/what-is-service-builder)
+should wait for the upgrade steps to be executed. By adding an `@Reference`
```

Comment:
Service Builder modules register a special upgrade step for "0.0.0" (You can check it in ModuleApplicationContextExtender._processInitialUpgrade).

In ModuleApplicationContextExtender._generateReleaseInfo you can see how we create a serviceDependency of the release.bundle.symbolic.name, the same dependency that we manually create for the Portlet in the Web modules.